File: src/IsbInterop/EDocuments/IEDocumentVersion.vb
Imports IsbInterop.DataTypes.Enumerable

Namespace EDocuments

  ''' <summary>
  ''' Версия электронного документа.
  ''' </summary>
  Public Interface IEDocumentVersion
    Inherits IIsbComObjectWrapper

    ''' <summary>
    ''' Текущее сосотяние версии.
    ''' </summary>
    ReadOnly Property CurrentState As TEDocumentVersionState

    ''' <summary>
    ''' Скрытая версия.
    ''' </summary>
    ReadOnly Property IsHidden As Boolean

    ''' <summary>
    ''' Номер версии.
    ''' </summary>
    ReadOnly Property Number As Integer

    ''' <summary>
    ''' Размер версии.
    ''' </summary>
    ReadOnly Property Size As Integer
  End Interface
End Namespace

File: src/IsbInterop.CSharp/EDocuments/IEDocumentVersion.cs
using IsbInterop.DataTypes.Enumerable;

namespace IsbInterop.EDocuments
{
  /// <summary>
  /// Версия электронного документа.
  /// </summary>
  public interface IEDocumentVersion : IBaseIsbObject
  {
    /// <summary>
    /// Текущее сосотяние версии.
    /// </summary>
    TEDocumentVersionState CurrentState { get; }

    /// <summary>
    /// Скрытая версия.
    /// </summary>
    bool IsHidden { get; }

    /// <summary>
    /// Номер версии.
    /// </summary>
    int Number { get; }

    /// <summary>
    /// Размер версии.
    /// </summary>
    int Size { get; }
  }
}
